citesphere/src/test/java/edu/asu/diging/citesphere/core/service/impl/AuthorityServiceTest.java

```diff
@@ -238,11 +310,297 @@ public void test_save() {
         entry.setName(name);
         entry.setUri(uri);
         Mockito.when(entryRepository.save(entry)).thenReturn(entry);
-        
+
         IAuthorityEntry actualEntry = managerToTest.save(entry);
         Mockito.verify(entryRepository).save(entry);
         Assert.assertEquals(entry.getId(), actualEntry.getId());
         Assert.assertEquals(entry.getName(), actualEntry.getName());
         Assert.assertEquals(entry.getUri(), actualEntry.getUri());
     }
+
+    @Test
+    public void test_findByName_onlyOnFirstName() {
+
+        Mockito.when(authorityRepository.findByUsernameAndNameContainingAndNameContainingOrderByName(user.getUsername(),
+                "", "Albert", paging)).thenReturn(entriesAlbert);
+
+        // Testing authority search with just first name
+        List<IAuthorityEntry> searchFirstnameAlbert = managerToTest.findByName(user, "", "Albert", page, pageSize);
+
+        Assert.assertEquals(3, searchFirstnameAlbert.size());
+        for (IAuthorityEntry entry : searchFirstnameAlbert) {
+            Assert.assertTrue(entry.getName().contains("Albert"));
+        }
+    }
+
+    @Test
+    public void test_findByName_onlyOnLastName() {
+
+        // removing authority who does not have 'Einstein' as last name
+        entriesAlbert.remove(1);
```

Comment:
I wouldn't do that. if someone else adds another test case and adds another entry, then this will break these test cases. I'd just create the lists in the methods where you need them, you can still create the entries themselves in the setup method.